### PR TITLE
Allow running Gazebo examples without joystick

### PR DESCRIPTION
**Pull request by alexdewar**
We could also just use a virtual joystick as a fallback, but in the meantime let's at least get the examples to run without a joystick to make life easier for @nitik1998.

### CODE REVIEW

File: examples/gazebo_tank/run.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/bin/sh
-trap 'kill %1' SIGINT
+trap 'kill %1' INT
```
Comment (neworderofjamie) [inline examples/gazebo_tank/run.sh:2]: What's the difference here? I don't think I've ever actually used ``trap`` in bash but googled examples seem to use ``SIGINT``

File: examples/gazebo_tank/gazebo_tank.cc [hunk 2]
```diff
@@ -33,38 +34,48 @@ int bobMain(int argc, char **argv)
...
+    }
 
-    do {
-        // Check for joystick events
-        if (!joystick.update() && (!display || !display->update())) {
-            // A small delay so we don't hog CPU
-            std::this_thread::sleep_for(5ms);
-        }
-    } while (!joystick.isPressed(HID::JButton::B));
+    if (!display && !joystick) {
```
Comment (neworderofjamie) [inline examples/gazebo_tank/gazebo_tank.cc:60]: What happens in either of these cases when you have a display but no joystick?

File: examples/gazebo_tank/run.sh (hunk 1 above)
Comment (alexdewar) [reply examples/gazebo_tank/run.sh:2]: Apparently they're equivalent, but the standard POSIX way of doing it is to use ``INT`` instead. Nitik is using zsh which doesn't like ``SIGINT``

File: examples/gazebo_tank/gazebo_tank.cc (hunk 2 above)
Comment (alexdewar) [reply examples/gazebo_tank/gazebo_tank.cc:60]: If there's a display but no joystick, then you'll have an ``OpenCV`` window showing the current view and you can terminate the program by closing it.

**Review by neworderofjamie — APPROVED** [review]

**Pull request merged**